rust/main/utils/run-locally/src/utils.rs

```diff
@@ -134,3 +134,15 @@ pub fn get_matching_lines(file: &File, search_strings: &[&str]) -> HashMap<Strin
     }
     matches
 }
+
+pub fn get_matched_lines(file: &File, search_string: &str) -> Vec<String> {
+    let reader = io::BufReader::new(file);
+    let mut lines = reader.lines();
+    let mut matched_lines = Vec::new();
+    while let Some(Ok(line)) = lines.next() {
+        if line.contains(search_string) {
+            matched_lines.push(line);
+        }
+    }
+    matched_lines
+}
```

Comment:
see my other comment first, since I think we don't need this function

the opening of the file, iteration and matching of a predicate is common to both this function and `get_matching_lines`, so we should reuse it as much as possible. You can refactor such that the generic logic works like a `fold`, and the arguments it takes are the initial value and the accumulation closure (exactly like `fold` does).

Reply:
yeah, that makes sense :+1: